src/layer.rs

```diff
@@ -80,6 +80,17 @@ impl LayerSet {
         self.layers.iter_mut().find(|l| &*l.name == name)
     }
 
+    /// Get a mutable reference to a layer, by name, or create it if it doesn't exist.
+    pub fn get_or_create_mut(&mut self, name: &str) -> &mut Layer {
```

Comment:
Assuming we aren't going to have a non-mut variant I might just call this `get_or_create`?